src/object/cli_shard.c

```diff
@@ -211,16 +241,31 @@ int dc_rw_cb_csum_verify(const struct rw_cb_args *rw_args)
 					  rw_args->shard_args->reasb_req);
 	for (i = 0; i < orw->orw_nr; i++) {
 		daos_iod_t		*iod = &iods[i];
+		daos_iod_t		 shard_iod = *iod;
+		d_sg_list_t		 shard_sgl = sgls[i];
 		struct dcs_iod_csums	*iod_csum = &iods_csums[i];
 		daos_iom_t		*map = &maps[i];
 
 		if (!csum_iod_is_supported(iod))
 			continue;
 
+		if (iod->iod_type == DAOS_IOD_ARRAY && oiods != NULL) {
```

Comment:
In what case would oiods be null?

Reply:
Now we only set it for EC obj, see the "shard_arg->oiods" assignment in shard_rw_prep().